pkg/util/acrtoken/acrtoken.go

```diff
@@ -98,9 +99,55 @@ func (m *manager) EnsureTokenAndPassword(ctx context.Context, rp *api.RegistryPr
 		return "", err
 	}
 
+	return m.generateTokenPassword(ctx, mgmtcontainerregistry.TokenPasswordNamePassword1, rp)
+}
+
+// RotateTokenPassword chooses either the unused token password or the token
+// password with the oldest creation date, generates a new password, and
+// then updates the registry profile with the newly generated password.
+func (m *manager) RotateTokenPassword(ctx context.Context, rp *api.RegistryProfile) error {
+	tokenProperties, err := m.tokens.GetTokenProperties(ctx, m.r.ResourceGroup, m.r.ResourceName, rp.Username)
+	if err != nil {
+		return err
+	}
+	tokenPasswords := *tokenProperties.Credentials.Passwords
+
+	var passwordToRenew mgmtcontainerregistry.TokenPasswordName
+	switch {
```

Comment:
We'll want a default case in here to protect us from unexpected lengths.

Reply:
@dem4gus, is this the doc that confirms one or two passwords can be created for a token? :- https://learn.microsoft.com/en-us/azure/container-registry/container-registry-repository-scoped-permissions#add-token-password

Reply:
That is one place, yes.